File: 0049-group-anagrams/0049-group-anagrams.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        HashMap<String, List<String>> ans = new HashMap<>();

        for(String s : strs){
            char[] c = s.toCharArray();
            Arrays.sort(c);
            String p = new String(c);

            if(!ans.containsKey(p)){
                ans.put(p, new ArrayList<>());
            }
            ans.get(p).add(s);
        }

        return new ArrayList<>(ans.values());
        
    }
}

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:

        freq = {}

        for s in strs:
            sorted_string = ''.join(sorted(s))

            if sorted_string in freq:
                freq[sorted_string].append(s)

            else:
                freq[sorted_string] = [s]

            


        return list(freq.values())

        